client.go

```diff
@@ -181,11 +178,37 @@ func populateClientConfig(config *Config) *Config {
 
 // establishSecureConnection returns as soon as the connection is secure (as opposed to forward-secure)
 func (c *client) establishSecureConnection() error {
+	if err := c.createNewSession(nil); err != nil {
+		return err
+	}
 	go c.listen()
 
+	var runErr error
+	errorChan := make(chan struct{})
+	go func() {
+		// session.run() returns as soon as the session is closed
+		for {
```

Comment:
Should we have some logic here to stop retrying after some number of tries?

Reply:
There must not be more than one try. I'll fix that.

Reply:
Can we just get rid of the for loop? I find it very non-intuitive if it just runs twice.

Reply:
Done. This code actually looks a lot nicer now.